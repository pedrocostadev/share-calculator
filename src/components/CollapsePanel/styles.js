import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    height: '100%',
  },
  expandedButtonContainer: {
    height: '5%',
  },
  collapsedButtonContainer: {
    height: '100%',
  },
  button: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#000000',
  },
  contentContainer: {
    height: '95%',
  },
});

const svgStyles = {
  chevron: {
    color: 'white',
  },
  expandedChevron: {
    rotation: 90,
  },
  collapsedChevron: {
    rotation: 270,
  },
};

export default { ...styles, ...svgStyles };
