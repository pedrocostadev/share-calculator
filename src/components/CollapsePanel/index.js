import * as React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, View } from 'react-native';

import Chevron from '../../svg/Chevron';
import styles from './styles';

const CollapsePanel = ({ isVisible, onPress, children }) => {
  const chevronStateStyle = isVisible ? styles.expandedChevron : styles.collapsedChevron;
  const chevronProps = { ...styles.chevron, ...chevronStateStyle };
  return (
    <View style={styles.container}>
      <View style={isVisible ? styles.expandedButtonContainer : styles.collapsedButtonContainer}>
        <TouchableOpacity testID="collapse-panel-button" style={styles.button} onPress={onPress}>
          <Chevron {...chevronProps} />
        </TouchableOpacity>
      </View>
      {isVisible && <View style={styles.contentContainer}>{children}</View>}
    </View>
  );
};

CollapsePanel.propTypes = {
  isVisible: PropTypes.bool.isRequired,
  onPress: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};

export default CollapsePanel;
