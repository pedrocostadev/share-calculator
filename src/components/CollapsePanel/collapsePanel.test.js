import React from 'react';
import { Text } from 'react-native';
import { render, cleanup, NativeTestEvent, fireEvent } from '@testing-library/react-native';

import CollapsePanel from './';

describe('<CollapsePanel />', () => {
  afterEach(cleanup);

  test('should render the children', async () => {
    const randomTitle = 'randomTitle';
    const onPress = jest.fn();
    const { queryAllByText, container } = render(
      <CollapsePanel isVisible onPress={onPress}>
        <Text>{randomTitle}</Text>
      </CollapsePanel>,
    );
    const ocurrences = await queryAllByText(randomTitle);
    expect(ocurrences).toHaveLength(1);
    expect(container).toMatchSnapshot();
  });
  test('should call onPress after click collapse button', async () => {
    const randomTitle = 'randomTitle';
    const onPress = jest.fn();
    const { queryAllByText, getByTestId } = render(
      <CollapsePanel isVisible onPress={onPress}>
        <Text>{randomTitle}</Text>
      </CollapsePanel>,
    );
    const ocurrences = await queryAllByText(randomTitle);
    expect(ocurrences).toHaveLength(1);
    expect(onPress).toHaveBeenCalledTimes(0);
    const collapseButton = getByTestId('collapse-panel-button');
    fireEvent(collapseButton, new NativeTestEvent('press'));
    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
