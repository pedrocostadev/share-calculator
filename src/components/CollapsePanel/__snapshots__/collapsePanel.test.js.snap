// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollapsePanel /> should render the children 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "height": "5%",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#000000",
            "display": "flex",
            "justifyContent": "center",
          }
        }
        testID="collapse-panel-button"
      >
        <RNSVGSvgView
          bbHeight="29"
          bbWidth="28"
          focusable={false}
          height="29"
          rotation={90}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              Object {
                "opacity": 1,
              },
              Object {
                "flex": 0,
                "height": 29,
                "width": 28,
              },
            ]
          }
          width="28"
        >
          <RNSVGGroup
            fill={4278190080}
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          >
            <RNSVGPath
              d="M12.586 12L9.293 8.707a1 1 0 1 1 1.414-1.414l4 4a1 1 0 0 1 0 1.414l-4 4a1 1 0 1 1-1.414-1.414L12.586 12z"
              fill={4294967295}
              fillOpacity={1}
              fillRule={1}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                  "stroke",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={4294967295}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "height": "95%",
        }
      }
    >
      <Text>
        randomTitle
      </Text>
    </View>
  </View>
</View>
`;
